src/highlighters.cc

```diff
@@ -998,6 +998,89 @@ struct TabulationHighlighter : Highlighter
     }
 };
 
+struct ShowMatchingHighlighter : Highlighter
+{
+    ShowMatchingHighlighter(int offset)
+      : Highlighter{HighlightPass::Move}, m_offset{std::move(offset)}
+      {}
+
+    static std::unique_ptr<Highlighter> create(HighlighterParameters params, Highlighter*)
+    {
+        static const ParameterDesc param_desc{
+            { { "offset", { true, "" } } },
+            ParameterDesc::Flags::None, 0, 0
+        };
+        ParametersParser parser(params, param_desc);    
+
+        int offset = parser.get_switch("offset").map(str_to_int).value_or(0);
+
+        return std::make_unique<ShowMatchingHighlighter>(
+            offset);
+    }
+private:
+        void do_highlight(HighlightContext context, DisplayBuffer& display_buffer, BufferRange) override
+        {
+            const Face face = context.context.faces()["MatchingChar"];
+            const auto& matching_pairs = context.context.options()["matching_pairs"].get<Vector<Codepoint, MemoryDomain::Options>>();
+            const auto range = display_buffer.range();
+            const auto& buffer = context.context.buffer();
+            for (auto& sel : context.context.selections())
+            {
+                auto pos = sel.cursor();
+                pos.column = pos.column-m_offset;
```

Comment:
This can get to negative columns which would have issues. It also wont work once you migrate to a CharCount for offset as we dont know how many bytes back we have to go.